include/clink/feature/one_hot_encoder.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2021 The Clink Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CLINK_FEATURE_ONE_HOT_ENCODER_H_
+#define CLINK_FEATURE_ONE_HOT_ENCODER_H_
+
+#include "clink/feature/proto/one_hot_encoder.pb.h"
+#include "clink/linalg/sparse_vector.h"
+#include "llvm/Support/Errc.h"
+#include "llvm/Support/Error.h"
+
+namespace clink {
+
+// A Model which encodes data into one-hot format.
+class OneHotEncoderModel {
+public:
+  // Default constructor.
+  OneHotEncoderModel() {}
+
+  // Move operations are supported.
+  OneHotEncoderModel(OneHotEncoderModel &&other) = default;
+  OneHotEncoderModel &operator=(OneHotEncoderModel &&other) = default;
+
+  // This class is copyable and assignable.
```

Comment:
OK. I'll make the change.

I previously set it to copyable because we have methods that return `llvm::Expected<OneHotEncoderModel>` so I need to use `llvm::Expected::get()` method to retrieve the model. Now that I changed it to pointer dereference and there is no need for it to be copyable.